Python/bltinmodule.c

```diff
@@ -1465,21 +1523,42 @@ map_reduce(mapobject *lz, PyObject *Py_UNUSED(ignored))
         PyTuple_SET_ITEM(args, i+1, Py_NewRef(it));
     }
 
+    if (lz->strict) {
+        return Py_BuildValue("ONO", Py_TYPE(lz), args, Py_True);
+    }
     return Py_BuildValue("ON", Py_TYPE(lz), args);
 }
 
+PyDoc_STRVAR(setstate_doc, "Set state information for unpickling.");
+
+static PyObject *
+map_setstate(mapobject *lz, PyObject *state)
+{
+    // For now we keep things strict, rather than using PyObject_IsTrue().
+    int strict = state == Py_True;
+    if (strict < 0) {
+        return NULL;
+    }
+    lz->strict = strict;
```

Comment:
```suggestion
    int strict = state == Py_True;
    lz->strict = strict;
```
This part of the code could never be reached right?